test/api/dependency/samples.go

```diff
@@ -0,0 +1,44 @@
+package dependency
+
+import (
+	"github.com/konveyor/tackle2-hub/api"
+)
+
+// Set of valid resources for tests and reuse.
+type TestCase struct {
+	Name         string
+	Dependency   api.Dependency
+	Application1 api.Application
+	Application2 api.Application
+}
+
+var (
+	dependency = api.Dependency{
+		To: api.Ref{
+			ID:   uint(1),
+			Name: "Alice",
+		},
+		From: api.Ref{
+			ID:   uint(2),
+			Name: "Bob",
+		},
+	}
+
+	aliceApplication = api.Application{
+		Name:        "Alice",
+		Description: "alice's application",
+	}
+	bobApplication = api.Application{
+		Name:        "Bob",
+		Description: "bob's application",
+	}
+
+	testCase = TestCase{
+		Name:         "Test",
+		Dependency:   dependency,
+		Application1: aliceApplication,
+		Application2: bobApplication,
+	}
+	tc      = []TestCase{testCase}
+	Samples = []api.Dependency{tc}
+)
```

Comment:
Definition of Dependencies (and likely also TestCases) will need to be updated. There is some limitation caused by api.Dependency keeping just Ref of Aplication, so we need store application elsewhere. I can see few options:
- Use TestCases struct that might even not contain the Dependency itself, but something like just ApplicationFrom and ApplicationTo, so the test will go through testcases and create ApplicationTo, ApplicationFrom and then dependency between these applications (the dependency doesn't have any special fields than the Application refs, so it should work well).
- Use api.Dependency model, that stores links to Application (not just Refs), see https://github.com/konveyor/tackle2-hub/blob/main/migration/v2/model/dependency.go#L12-L18 (yes, we do have slightly different models for database and API). The the Samples array would be array of these model.Dependency and test could ceate both Applications and "convert" the "convert" the model.Dependency to api.Dependency using method With https://github.com/konveyor/tackle2-hub/blob/main/api/dependency.go#L157-L161
- It would great if you can try both ways (or maybe find something else if you'd like it more than suggestions above) and we can discuss how it works if needed.

Reply:
I think I have some idea regarding the first option ( I shall try creating the TestCases without the dependency and then after try to inject a sample dependency in the Samples array). And pass similar structure to the `api_test.go` . An example of structure in `api.test.go` according to what I can undestand can be 
```
applicationFrom := api.Application{
				Name:        r.From.Name,
				Description: "From application",
}

applicationTo := api.Application{
			       Name:        r.To.Name,
				Description: "To application",
}
dependency := api.Dependency{
				From: api.Ref{
					ID: applicationFrom.ID,
				},
				To: api.Ref{
					ID: applicationTo.ID,
				},
}
```
and then pass like this `err := Dependency.Create(&dependency)`. Am i going towards the right direction ?

Reply:
> Definition of Dependencies (and likely also TestCases) will need to be updated. There is some limitation caused by api.Dependency keeping just Ref of Aplication, so we need store application elsewhere. I can see few options:
> 
> * Use TestCases struct that might even not contain the Dependency itself, but something like just ApplicationFrom and ApplicationTo, so the test will go through testcases and create ApplicationTo, ApplicationFrom and then dependency between these applications (the dependency doesn't have any special fields than the Application refs, so it should work well).
> * Use model.Dependency model, that stores links to Application (not just Refs), see https://github.com/konveyor/tackle2-hub/blob/main/migration/v2/model/dependency.go#L12-L18 (yes, we do have slightly different models for database and API). The Samples array would be array of these model.Dependency and test could create both Applications and "convert" the model.Dependency to api.Dependency using method With() https://github.com/konveyor/tackle2-hub/blob/main/api/dependency.go#L157-L161
> * It would great if you can try both ways (or maybe find something else if you'd like it more than suggestions above) and we can discuss how it works if needed.

I think 1st option is easier to implement because it involves less complexities and 2nd option is is less intuitive which can be easily dealt by the 1st. What's your say on that?

Reply:
I have a thought process of injecting dependency withdependecy-less TestCase in Samples[]. and then use From/To in the `api_test.go` to access them in the `dependency` to be used by various CRUD functions.

Reply:
I'm not sure I fully understand, but sounds good.